lib/beaker/hypervisor/vagrant.rb

```diff
@@ -201,7 +201,8 @@ def set_ssh_config host, user
 
         f.write(ssh_config)
         f.rewind
-        host['ssh'] = {:config => f.path()}
+
+        host['ssh'] = host['ssh'].merge(Net::SSH.configuration_for(host['ip'], f.path))
```

Comment:
Thank for your contribution and testing!

I think this is an elegant solution. From what I see in the code these files are based on the command `vagrant ssh-config` and modified to use `host['ip']`, `host['user']`, and set `IdentitiesOnly no`. So this would replace the `host['ssh']['user']` if I'm understanding it well.

If that is the correct behavior, can we document that somewhere?

Lastly, this change removes the previous `:config` key,  my understanding is that it is needed for the rsync method, so let's keep it in order not to break that method.

```suggestion
        host['ssh'][:config] = f.path()
        host['ssh'] = host['ssh'].merge(Net::SSH.configuration_for(host['ip'], f.path))
```